facepy/signed_request.py

```diff
@@ -72,15 +79,14 @@ def __init__(self, signed_request=None, application_secret_key=None, application
         )
 
     def fetch_user_data_and_token(self):
-        from urlparse import parse_qs
         from . import GraphAPI, get_application_access_token
 
         app_token = get_application_access_token(self.application_id, self.application_secret_key, api_version=self.api_version)
         graph = GraphAPI(app_token, version=self.api_version)
 
         qs = graph.get('oauth/access_token', code=self.raw['code'], redirect_uri='', client_id=self.application_id, client_secret=self.application_secret_key)
-        self.raw['oauth_token'] = parse_qs(qs)['access_token'][0]
-        self.raw['expires'] = time.time() + int(parse_qs(qs)['expires'][0])
+        self.raw['oauth_token'] = qs['access_token']
+        self.raw['expires'] = time.time() + qs['expires_in']
```

Comment:
If there are still older API versions that cause `get` to return a string instead of a dict, I could add the try/except that’s already used elsewhere in the code.  Maybe the `get` method (via `_parse`) should be responsible though, instead of all the client methods, and maybe you want unit tests before this is changed, so I did the simplest fix that worked here.